src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -817,9 +819,12 @@ await using (var stream = GetInputStream(decompressionStream, snapshotEncryption
 
             result.Files.CurrentFileName = null;
 
-            long totalExecutedCommands = 0;
-
             //when restoring from a backup, the database doesn't exist yet and we cannot rely on the DocumentDatabase to execute the database cluster transaction commands
+            ExecuteClusterTransactions(database, onProgress, result, ref totalExecutedCommands);
```

Comment:
@grisha-kotler I wonder if this should be part of the smuggler and not external to it.
What happened when we do normal import? (and not via restore)

Reply:
so should we call `ExecuteClusterTransactions` after import as well?

Reply:
There is already a dedicated thread that is running the cluster transactions.
We can just wait for the last imported cluster transaction to be applied (which we cannot do for backup).

Reply:
okay so we can rely on `BackupKind` for that purpose?

Reply:
Was it handled? @shaharhikri?

Reply:
Yes